### PR TITLE
Cake issues ruby 2.3

### PR DESCRIPTION
**Pull request by avakarev**
This PR brings latest changes from master regarding upgrading to ruby 2.3.0
Also specs was broken, so this is also fixing specs.

There is one rubocop issue left which I am not sure how to fix, please see comments.

### CODE REVIEW

**Comment by dziemba** [comment]
Isn't this in master already?

File: lib/soapy_cake/client.rb [hunk 1]
```diff
@@ -22,7 +22,7 @@ def xml_response?
     attr_reader :domain, :api_key, :time_converter, :opts, :logger
 
     def run(request)
```
Comment (avakarev) [inline lib/soapy_cake/client.rb]: Rubocop failing on run method with:
```bash
$:> be rake rubocop
Running RuboCop...
Inspecting 32 files
........C.......................

Offenses:

lib/soapy_cake/client.rb:24:5: C: Metrics/AbcSize: Assignment Branch Condition size for run is too high. [18.68/15]
    def run(request)
    ^^^

32 files inspected, 1 offense detected
RuboCop failed!
```
I am still looking into it.

Comment (avakarev) [inline lib/soapy_cake/client.rb]: Yes, it is. But we still use cake-issues branch in BO. So, I've back-merged master to keep this PR mergeable with the cake-issues branch.

Comment (dziemba) [inline lib/soapy_cake/client.rb]: Hmm, cake-issues will not be needed anymore after tomorrow (?) when the bucket_cake PR is merged.

Comment (avakarev) [inline lib/soapy_cake/client.rb]: Didn't knew this. Then, I will just close this PR.

**Pull request closed without merging**